pandas/core/series.py

```diff
@@ -3814,9 +3819,13 @@ def explode(self) -> "Series":
 
         values, counts = reshape.explode(np.asarray(self.array))
 
-        result = self._constructor(
-            values, index=self.index.repeat(counts), name=self.name
-        )
+        if ignore_index:
+            result = self._constructor(values, index=range(len(values)), name=self.name)
+        else:
+            result = self._constructor(
```

Comment:
actually  can you do the index construction first, e.g. respecting `ignore_index`, then construct the result

Reply:
Implemented this, if I understood you correctly.